pkg/core/resources/apis/meshexternalservice/api/v1alpha1/meshexternalservice.go

```diff
@@ -2,17 +2,155 @@
 package v1alpha1
 
 import (
-	common_api "github.com/kumahq/kuma/api/common/v1alpha1"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+
+	"github.com/kumahq/kuma/api/common/v1alpha1"
 )
 
 // MeshExternalService
 // +kuma:policy:is_policy=false
 // +kuma:policy:has_status=true
 type MeshExternalService struct {
-	// TargetRef is a reference to the resource the policy takes an effect on.
-	// The resource could be either a real store object or virtual resource
-	// defined in-place.
-	TargetRef common_api.TargetRef `json:"targetRef"`
+	// Match defines traffic that should be routed through the sidecar.
+	Match Match `json:"match"`
+	// Extension struct for a plugin configuration, in the presence of an extension `endpoints` and `tls` are not required anymore - it's up to the extension to validate them independently.
+	Extension Extension `json:"extension,omitempty"`
+	// Endpoints defines a list of destinations to send traffic to.
+	Endpoints []Endpoint `json:"endpoints,omitempty"`
+	// Tls provides a TLS configuration when proxy is resposible for a TLS origination
+	Tls Tls `json:"tls,omitempty"`
+}
+
+// +kubebuilder:validation:Enum=HostnameGenerator
+type MatchType string
+
+// +kubebuilder:validation:Enum=tcp;grpc;http;http2
+type ProtocolType string
+
+type Match struct {
+	// Type of the match, only `HostnameGenerator` is available at the moment.
+	// +kubebuilder:default=HostnameGenerator
+	Type MatchType `json:"type,omitempty"`
+	// Port defines a port to which a user does request.
+	Port int `json:"port"`
+	// Protocol defines a protocol of the communication. Possible values: `tcp`, `grpc`, `http`, `http2`.
+	// +kubebuilder:default=tcp
+	Protocol ProtocolType `json:"protocol,omitempty"`
+}
+
+type Extension struct {
+	// Type of the extension.
+	Type string `json:"type"`
+	// Config freeform configuration for the extension.
+	Config *apiextensionsv1.JSON `json:"config"`
+}
+
+// +kubebuilder:validation:Minimum=1
+// +kubebuilder:validation:Maximum=65535
+type EndpointPort int
+
+type Endpoint struct {
+	// Address defines an address to which a user want to send a request. Is possible to provide `domain`, `ip` and `unix` sockets.
+	// +kubebuilder:example="127.0.0.1"
+	// +kubebuilder:example="example.com"
+	// +kubebuilder:example="unix:///tmp/example.sock"
+	// +kubebuilder:validation:MinLength=1
+	Address string `json:"address"`
+	// Port of the endpoint
+	Port EndpointPort `json:"port"`
+}
+
+type Tls struct {
+	// Enabled defines if proxy should originate TLS.
+	// +kubebuilder:default=false
+	Enabled bool `json:"enabled,omitempty"`
+	// Version section for providing version specification.
+	Version TlsVersion `json:"version"`
```

Comment:
There's a default here too no?

Reply:
yeah, it is here https://github.com/kumahq/kuma/blob/9bd764a5d14da11044517804c6f21266e4525f7e/pkg/core/resources/apis/meshexternalservice/api/v1alpha1/meshexternalservice.go#L83

Reply:
does it make too much difference if it's on the object vs on the property?